Plugin/FileSystemWorkspace/clFileSystemWorkspace.cpp

```diff
@@ -999,8 +1005,30 @@ void clFileSystemWorkspace::OnDebug(clDebugEvent& event)
     clDebugEvent eventStarted(wxEVT_DEBUG_STARTED);
     eventStarted.SetClientData(&session_info);
     EventNotifier::Get()->ProcessEvent(eventStarted);
-    // Now run the debuggee
-    dbgr->Run(args, "");
+    
+    if (dbgr->GetIsRemoteDebugging()) {
+        // debugging remote target
+        wxString comm;
+        wxString host = GetConfig()->GetDebuggerRemoteHost();
+        wxString port = GetConfig()->GetDebuggerRemotePort();
+        
+        comm << host;
+
+        host = host.Trim().Trim(false);
```

Comment:
You probably invert those 2 lines. else modifying host is useless.

Reply:
Yes, you right. It was blindly copied from LiteEditor/manager.cpp (around line 2238) :)
So this "bug" already present in CodeLite (https://github.com/eranif/codelite/blob/287ff78fa959f6f4491dea73ea1dc5bf92710cd3/LiteEditor/manager.cpp#L2080)